File: backend/app/schemas/forum_import.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class ForumImportSettingsUpdate(BaseModel):
    """Forum import settings update schema"""
    forum_url: Optional[str] = None
    forum_username: Optional[str] = None
    forum_password: Optional[str] = None
    forum_subforum_id: Optional[int] = None


class ForumImportStatusResponse(BaseModel):
    """Forum import status response schema"""
    id: int
    is_running: bool
    status: str
    total_topics: int
    processed_topics: int
    successful_topics: int
    failed_topics: int
    current_topic_title: Optional[str] = None
    current_operation: Optional[str] = None
    started_at: Optional[datetime] = None
    finished_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None
    last_error: Optional[str] = None
    forum_url: Optional[str] = None
    subforum_id: Optional[int] = None

    class Config:
        from_attributes = True


class ForumImportStartRequest(BaseModel):
    """Forum import start request schema"""
    forum_url: str
    forum_username: str
    forum_password: str
    subforum_id: int
    max_topics: int = 10
    api_email: str
    api_password: str

File: frontend/src/api/forumImport.ts
import { api } from './client';

export interface ForumImportStatus {
  id: number;
  is_running: boolean;
  status: string;
  total_topics: number;
  processed_topics: number;
  successful_topics: number;
  failed_topics: number;
  current_topic_title?: string;
  current_operation?: string;
  started_at?: string;
  finished_at?: string;
  updated_at?: string;
  last_error?: string;
  forum_url?: string;
  subforum_id?: number;
}

export interface ForumImportSettings {
  forum_url?: string;
  forum_username?: string;
  forum_password?: string;
  forum_subforum_id?: number;
}

export interface ForumImportStartRequest {
  forum_url: string;
  forum_username: string;
  forum_password: string;
  subforum_id: number;
  max_topics: number;
  api_email: string;
  api_password: string;
}

export const forumImportApi = {
  getStatus: async (): Promise<ForumImportStatus> => {
    const response = await api.get('/forum-import/status');
    return response.data;
  },

  startImport: async (request: ForumImportStartRequest): Promise<ForumImportStatus> => {
    const response = await api.post('/forum-import/start', request);
    return response.data;
  },

  stopImport: async (): Promise<ForumImportStatus> => {
    const response = await api.post('/forum-import/stop');
    return response.data;
  },

  getSettings: async (): Promise<ForumImportSettings> => {
    const response = await api.get('/forum-import/settings');
    return response.data;
  },

  updateSettings: async (settings: Partial<ForumImportSettings>): Promise<ForumImportSettings> => {
    const response = await api.put('/forum-import/settings', settings);
    return response.data;
  },
};
